From e279679dd7308efebe004252d168d7308f3b99ce Mon Sep 17 00:00:00 2001
From: Daniel Poelzleithner <poelzleithner@b1-systems.de>
Date: Wed, 25 Oct 2023 14:26:49 +0200
Subject: [PATCH 1/2] Add .desktop file and fix install

Cleanup CMake file to not install license in
bin folder, but proper XDG path.

Add .desktop file so the application is found
by the menu system
---
 CMakeLists.txt      | 11 ++++++++++-
 share/kemai.desktop |  9 +++++++++
 2 files changed, 19 insertions(+), 1 deletion(-)
 create mode 100644 share/kemai.desktop

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9593f29..d5cfe55 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,9 +76,19 @@ elseif (APPLE)
     #    set(CMAKE_OSX_DEPLOYMENT_TARGET 10.12)
     set(DEPLOY_DIR .)
     set(CPACK_GENERATOR DragNDrop)
+    install(FILES LICENSE.txt DESTINATION ${DEPLOY_DIR})
 else ()
+    include(GNUInstallDirs)
     set(OS_BUNDLE)
     set(DEPLOY_DIR bin)
+    # install(FILES share/kemai.desktop DESTINATION ${)
+    install(FILES share/kemai.desktop
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/applications
+    )
+    install(FILES src/resources/icons/kimai.png
+        DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/256x256/apps/
+    )
+    install(FILES LICENSE.txt DESTINATION ${CMAKE_INSTALL_DOCDIR}/)
 endif ()
 
 # Project code
@@ -86,7 +96,6 @@ add_subdirectory(src)
 
 # Install targets and files
 install(TARGETS Kemai DESTINATION ${DEPLOY_DIR})
-install(FILES LICENSE.txt DESTINATION ${DEPLOY_DIR})
 
 if (WIN32)
     windeployqt(Kemai ${DEPLOY_DIR})
diff --git a/share/kemai.desktop b/share/kemai.desktop
new file mode 100644
index 0000000..df76fc5
--- /dev/null
+++ b/share/kemai.desktop
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Name=Kemai
+Comment=Kimai timetracking frontend
+TryExec=Kemai
+Exec=Kemai
+Terminal=false
+Categories=Office;Utility;
+Icon=kimai
-- 
2.40.1

